cmd/cloudeventingress/main.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client"
+	"github.com/knative/eventing-sources/pkg/kncloudevents"
+)
+
+const (
+	// Sink to transform CloudEvent data.
+	envSinkURI = "SINK_URI"
+)
+
+var (
+	sink     string
+	ceClient client.Client
+)
+
+func init() {
+	flag.StringVar(&sink, "sink", "", "uri to send events to")
+}
+
+func receive(event cloudevents.Event) {
+	ec := event.Context.AsV02()
+	log.Printf("CloudEvent:\n%s", event)
+	log.Printf("[%s] %s %s: ", ec.Time, event.DataContentType(), ec.Source.String())
+
+	//send out
+	if _, err := ceClient.Send(context.TODO(), event); err != nil {
+		log.Fatalf("failed to send cloudevent\n%s", err)
+	}
+}
+
+func main() {
+	ctx := context.TODO()
+
+	flag.Parse()
+
+	if sink == "" {
+		log.Fatalf("no sink provided")
+	}
+
+	ceClient, err := kncloudevents.NewDefaultClient()
+	if err != nil {
+		log.Fatalf("failed to create client:%s", err)
+	}
+
+	if err := ceClient.StartReceiver(ctx, receive); err != nil {
```

Comment:
StartReceiver is now blocking in CloudEvents SDK >=0.4.0 so you can just do this now:

```
log.Fatalf(ceClient.StartReceiver(ctx, receive))
```